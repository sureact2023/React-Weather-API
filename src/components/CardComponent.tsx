import React, {FC} from 'react'
import Card from '@mui/material/Card';

interface iCardComponent{
    icon: JSX.Element | JSX.Element[],
    data: string
}

const CardComponent: FC<iCardComponent> = ({data, icon}) => {
  return (
    <Card>
      {icon}
      {data}
    </Card>
  )
}

export default CardComponent
