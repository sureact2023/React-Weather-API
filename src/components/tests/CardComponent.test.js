import { render, screen } from "@testing-library/react";
import CardComponent from "../CardComponent";
import WaterIcon from '@mui/icons-material/Water';

// YOU NEED TO SEPARATE 2 THINGS
// DATA & COMPONENT

describe("Testing the CardComponent",()=>{

    it('Snapshop testing', ()=>{
        const {container} = render(<CardComponent data="hello" icon={<WaterIcon/>}/>)
        expect(container).toMatchSnapshot()
    })

    it('testing to check if the component renders correctly on screen or not', ()=>{
        const {container} = render(<CardComponent data="hello" icon={<WaterIcon/>}/>)
        expect(screen.getByText('hello')).toBeTruthy()
    })
})