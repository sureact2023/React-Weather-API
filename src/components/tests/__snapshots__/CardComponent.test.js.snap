// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the CardComponent Snapshop testing 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="WaterIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M21.98 14H22h-.02zM5.35 13c1.19 0 1.42 1 3.33 1 1.95 0 2.09-1 3.33-1 1.19 0 1.42 1 3.33 1 1.95 0 2.09-1 3.33-1 1.19 0 1.4.98 3.31 1v-2c-1.19 0-1.42-1-3.33-1-1.95 0-2.09 1-3.33 1-1.19 0-1.42-1-3.33-1-1.95 0-2.09 1-3.33 1-1.19 0-1.42-1-3.33-1-1.95 0-2.09 1-3.33 1v2c1.9 0 2.17-1 3.35-1zm13.32 2c-1.95 0-2.09 1-3.33 1-1.19 0-1.42-1-3.33-1-1.95 0-2.1 1-3.34 1-1.24 0-1.38-1-3.33-1-1.95 0-2.1 1-3.34 1v2c1.95 0 2.11-1 3.34-1 1.24 0 1.38 1 3.33 1 1.95 0 2.1-1 3.34-1 1.19 0 1.42 1 3.33 1 1.94 0 2.09-1 3.33-1 1.19 0 1.42 1 3.33 1v-2c-1.24 0-1.38-1-3.33-1zM5.35 9c1.19 0 1.42 1 3.33 1 1.95 0 2.09-1 3.33-1 1.19 0 1.42 1 3.33 1 1.95 0 2.09-1 3.33-1 1.19 0 1.4.98 3.31 1V8c-1.19 0-1.42-1-3.33-1-1.95 0-2.09 1-3.33 1-1.19 0-1.42-1-3.33-1-1.95 0-2.09 1-3.33 1-1.19 0-1.42-1-3.33-1C3.38 7 3.24 8 2 8v2c1.9 0 2.17-1 3.35-1z"
      />
    </svg>
    hello
  </div>
</div>
`;
